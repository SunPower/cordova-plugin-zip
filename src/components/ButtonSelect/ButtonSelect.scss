@import "../../styles/variables";

.stor {
  &.tags:not(:last-child) {
    margin-bottom: 0;
  }
  &.tags:last-child {
    margin-bottom: 0;
  }
  .tag {
    background: $dark;

    cursor: pointer;

    &:disabled span {
      color: $gray-text-default;
    }

    &:not(body):not(:first-of-type) {
      padding-left: 7px;
      padding-right: 10px;
      padding-top: 0;
    }

    &:first-of-type {
      border-top-left-radius: 20px;
      border-bottom-left-radius: 20px;
      border: 1px solid $navbar-white;
    }
    &:last-of-type {
      border-top-right-radius: 20px;
      border-bottom-right-radius: 20px;
      border: 1px solid $navbar-white;
      border-left: none;
      span {
        font-size: $font-size-huge;
        line-height: 0;
      }
    }
  }
}
