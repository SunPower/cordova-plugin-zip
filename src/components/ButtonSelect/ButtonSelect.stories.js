import React from 'react'

import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'

import ButtonSelect from './index'

storiesOf('ButtonSelect', module)
  .add('Simple', () => (
    <div className="is-flex level mt-20 mb-20 mr-20 ml-20">
      <ButtonSelect
        onAction={action('show something please')}
        label="Start Date"
      />
      <ButtonSelect
        onAction={action('show something please')}
        label="End Date"
      />
    </div>
  ))
  .add('With a value', () => (
    <div className="is-flex level mt-20 mb-20 mr-20 ml-20">
      <ButtonSelect
        onAction={action('did you just clicked?')}
        onClear={action("Let's clear the value later")}
        value="08/22/1990"
      />
    </div>
  ))
