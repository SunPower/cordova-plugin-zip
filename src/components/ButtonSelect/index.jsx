import React from 'react'

import './ButtonSelect.scss'

function ButtonSelect(props) {
  const { label, value, onAction, onClear } = props

  return (
    <div className="stor tags has-addons">
      <button className="tag has-text-white is-size-6" onClick={onAction}>
        {value || label}
      </button>

      <button
        className="tag has-text-white is-size-6 has-text-primary"
        onClick={onClear}
        disabled={!value}
      >
        <span>&times;</span>
      </button>
      {props.children}
    </div>
  )
}

export default ButtonSelect
