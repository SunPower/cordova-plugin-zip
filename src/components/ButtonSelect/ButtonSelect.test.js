import React from 'react'
import { shallow } from 'enzyme'

import ButtonSelect from '.'

describe('ButtonSelect Component', () => {
  it('renders without crashing', () => {
    const component = shallow(<ButtonSelect />)
    expect(component).toMatchSnapshot()
  })
})
