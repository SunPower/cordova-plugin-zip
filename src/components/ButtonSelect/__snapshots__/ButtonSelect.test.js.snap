// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonSelect Component renders without crashing 1`] = `
<div
  className="stor tags has-addons"
>
  <button
    className="tag has-text-white is-size-6"
  />
  <button
    className="tag has-text-white is-size-6 has-text-primary"
    disabled={true}
  >
    <span>
      Ã—
    </span>
  </button>
</div>
`;
